dataset/datasetGen.py

```diff
@@ -9,31 +10,52 @@ def Tag_To_Url(base: str, tag: str, page: int):
         content = response.read()
     tag_url = findall(r'<h3><a href="(.*?)" class', str(content))[1:]
     tag_url = [f'{base}{link}' for link in tag_url]
-    #return list of urls for that tag
+    # return list of urls for that tag
     yield tag_url
 
 
-def Link_To_JSON(urls: list):
-    url = urls[0]
+def Url_To_Data(url: str):
```

Comment:
Wouldn't it be great if we can split `url_to_data()` into small methods. This violate [SLAP](http://principles-wiki.net/principles:single_level_of_abstraction)

Reply:
Actually, that sounds better than having all the operations be done under one hood. I'll look into what can be moved around.